src/shogun/regression/KRRNystrom.h

```diff
@@ -0,0 +1,134 @@
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * The views and conclusions contained in the software and documentation are those
+ * of the authors and should not be interpreted as representing official policies,
+ * either expressed or implied, of the Shogun Development Team.
+ */
+
+#ifndef _KRRNYSTROM_H__
+#define _KRRNYSTROM_H__
+
+#include <shogun/regression/KernelRidgeRegression.h>
+
+namespace shogun {
+
+/** @brief Class KRRNystrom implements the Nyström method for kernel ridge
+ * regression, using a low-rank approximation to the kernel matrix.
+ *
+ * The method is equivalent to ordinary kernel ridge regression, but through a
+ * low-rank approximation to the original kernel matrix, the full kernel matrix
+ * does not need to be computed, and the resulting system of equations for the
+ * alphas is cheaper to solve.
+ *
+ * Given a data matrix X with n training examples along the rows, one chooses
+ * a subset m of the training example and projects all training examples upon
+ * these. This yields the following approximation to the kernel matrix
+ *
+ * \f[
+ * K \approx K_{n,m}K^+_{m,m}K_{m,n}
+ * ]
+ * where \f$K_{n,m}\f$ is a submatrix containing all n rows and the m columns
+ * corresponding to the m chosen training examples, \f$K_{m,n}\f$ is its
+ * transpose and \f$K_{m,m} is the submatrix with the m rows and columns
+ * corresponding to the training examples chosen. \f$+\f$ indicates the
+ * Moore-Penrose pseudoinverse.
+ *
+ * The resulting system of equations for the alphas can then be solved through
+ * the Woodbury formula, which gives the following equation for the alphas
+ *
+ * \f[
+ * {\bf \alpha} = \frac{1}{\tau} ({\bf y} - K_{n,m} (\tau K_{m,m} + K_{m,n}K_{n,m})^+K_{m,n} {\bf y}
+ * ]
```

Comment:
Maybe state that there are multiple ways to select columns. And that overloading the "subsample_indices" method can alter that